app/services/keys/query_service.rb

```diff
@@ -65,7 +65,7 @@ def query_local_devices!
   def query_remote_devices!
     return if @account.devices_url.blank?
 
-    json = fetch_resource(@account.devices_url)
+    json = fetch_resource(@account.devices_url, @account)
```

Comment:
That's still wrong, `@account` is the account you're fetching the keys of, not an account you can perform a query on behalf of.

What issue are you trying to fix?